specs/clipboard-api/clipboard.idl

```diff
@@ -0,0 +1,426 @@
+
+    Windows.Foundation.Collections.IVectorView<String> AvailableFormats { get; };
+    Boolean Contains(String formatId);
+
+    Windows.Foundation.IAsyncOperation<Object> GetDataAsync(String formatId);
+
+    [method_name("GetTextAsync")]
+    Windows.Foundation.IAsyncOperation<String> GetTextAsync();
+    [method_name("GetCustomTextAsync")]
+    Windows.Foundation.IAsyncOperation<String> GetTextAsync(String formatId);
+    Windows.Foundation.IAsyncOperation<String> GetHtmlFormatAsync();
+    Windows.Foundation.IAsyncOperation<String> GetRtfAsync();
+
+    Windows.Foundation.IAsyncOperation<Windows.Foundation.Uri> GetApplicationLinkAsync();
+    Windows.Foundation.IAsyncOperation<Windows.Foundation.Uri> GetWebLinkAsync();
+
+    Windows.Foundation.IAsyncOperation<Windows.Storage.Streams.RandomAccessStreamReference> GetBitmapAsync();
+
+    Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IVectorView<Windows.Storage.IStorageItem> > GetStorageItemsAsync();
+
+    Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IMapView<String, Windows.Storage.Streams.RandomAccessStreamReference> > GetResourceMapAsync();
+
+    [method_name("RequestAccessAsync")]
+    Windows.Foundation.IAsyncOperation<Windows.Security.EnterpriseData.ProtectionPolicyEvaluationResult> RequestAccessAsync();
+    [method_name("RequestAccessWithEnterpriseIdAsync")]
+    Windows.Foundation.IAsyncOperation<Windows.Security.EnterpriseData.ProtectionPolicyEvaluationResult> RequestAccessAsync(String enterpriseId);
+    Windows.Security.EnterpriseData.ProtectionPolicyEvaluationResult UnlockAndAssumeEnterpriseIdentity();
+
+    // FIXME: poorly documented on docs.microsoft.com - what does it do? why is in a View class?
+    void SetAcceptedFormatId(String formatId);
+  }
+
+  /// <summary>
+  /// A set of metadata properties for a <see cref="DataPackage"/>.
+  /// Contains strongly-typed WinRT properties for common descriptors
+  /// of the data or source app, but also allows setting arbitrary key-value mappings.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  runtimeclass DataPackagePropertySet :
+    [default] IDataPackagePropertySet
+    , Windows.Foundation.Collections.IMap<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  { }
+
+  [contract(ClipboardContract, 1.0)]
+  [exclusiveto(DataPackagePropertySet)]
+  [hasvariant]
+  interface IDataPackagePropertySet requires
+    Windows.Foundation.Collections.IMap<String, Object>
+    , Windows.Foundation.Collections.IIterable<Windows.Foundation.Collections.IKeyValuePair<String, Object> >
+  {
+    String Title;
+    String Description;
+    Windows.Storage.Streams.IRandomAccessStreamReference Thumbnail;
+    Windows.Foundation.Collections.IVector<String> FileTypes { get; };
+    String ApplicationName;
+    Windows.Foundation.Uri ApplicationListingUri;

```

Comment:
These DataPackagePropertySet properties assume that the DataPackage comes from an AppX/MSIX-packaged app. That was almost always true when this API was in UWP. Do we want to encode that assumption in the undocked version?